File: cdk_integration_tests/src/typescript/TransferServerIsPublic/fail__1__.ts
import { App, Stack } from 'aws-cdk-lib';
import * as transfer from 'aws-cdk-lib/aws-transfer';
import { Construct } from 'constructs';

class MyTransferServerStack extends Stack {
  constructor(scope: Construct, id: string, props?: {}) {
    super(scope, id, props);

    // Define Transfer Server with EndpointType set to a custom value
    new transfer.CfnServer(this, 'MyTransferServer', {
      endpointType: 'abc', // Replace 'abc' with your endpoint type
      // Other properties as needed for your Transfer Server
    });
  }
}

const app = new App();
new MyTransferServerStack(app, "MyTransferServerStack");
app.synth();

File: cdk_integration_tests/src/python/TransferServerIsPublic/fail__1__.py
from aws_cdk import core
from aws_cdk import aws_transfer as transfer

class MyTransferServerStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define Transfer Server with EndpointType set to VPC
        transfer.CfnServer(
            self, 'MyTransferServer',
            endpoint_type='abc',
            # Other properties as needed for your Transfer Server
        )

app = core.App()
MyTransferServerStack(app, "MyTransferServerStack")
app.synth()